app/models/miq_queue.rb

```diff
@@ -117,6 +117,9 @@ def self.put(options)
     options[:task_id]      = $_miq_worker_current_msg.try(:task_id) unless options.key?(:task_id)
     options[:role]         = options[:role].to_s unless options[:role].nil?
 
+    # Let's deprecate supplying non-array args soon
+    options[:args] = [options[:args]] if options[:args] && !options[:args].kind_of?(Array)
```

Comment:
@jrafanie So, this forces non Array `args` into an array. What about when the `args` are retrieved by the client? Will they get an array when they expect something else?